test/cpu_reduce_util.hpp

```diff
@@ -29,19 +29,177 @@
 #include <half.hpp>
 #include <limits>
 #include <cmath>
+#include <cassert>
+#include <stdexcept>
+#include <string>
 #include <miopen/miopen.h>
 #include <miopen/reduce_common.hpp>
 
 #include <miopen/bfloat16.hpp>
 
 namespace reduce {
 
+template <typename T>
+static inline bool IsNan(T x)
+{
+    // C++ isnan() is used for float, double and half_float::half
+    return (std::isnan(x));
+};
+
+template <typename T>
+static inline bool float_equal_one(T x)
+{
+    (void)x;
+    static_assert(static_cast<T>(0), "float_equal_one() is not implemented for this data type");
+    return false;
+};
```

Comment:
```suggestion
template <typename T>
static inline bool float_equal_one(T);
```
No definition given for a static function, so build will fail at compilation step.

Reply:
Yes, Removed the default function template definition.